### PR TITLE
Shrooms overhaul

### LINKED ISSUE #19695: Mushroom effects are spoilered by crafting recipes helper
Not sure if this is intentional but "generic" mushrooms are described with the same warning no matter what they are. However, the "You could use it to craft" section gives it away. See these two examples:

image
image

### PR DESCRIPTION
**Pull request by Night-Pryanik**
Resolves #19695.

- Removed hallucinogenic and poison mushrooms from a list potential forage results and from pizza parlor's fridge spawn;
- Foraging can now yield only ordinary mushrooms with poison or hallu flags;
- Eating poison or hallu shrooms will poison you or cause hallucinations on you, respectively;
- Crafting from hallu shrooms will result in hallu comestibles. Crafting from poison shrooms will result in safe comestible (let's presume proper cooking removes all toxins and poisons from a mushroom), except for dried poisonous mushrooms and poisonous seeds, which will still poison you;
- Poisonous and hallucinogenic mushrooms will be shown as `(poisonous)` and `(hallucinogenic)` respectively in the inventory if one has high survival levels;
- Lowered dried mushrooms charges from 4 to 1, lowered required mushrooms to craft them from 4 to 1 too;
- Migrated poison mushroom and seeds, hallu mushrooms and seeds to the ordinary variants;
- Removed separate hallu mushrooms recipe, removed separate poison and hallu shrooms' seeds recipes;
- Removed `FUN_HALLU` iuse function from `iuse.cpp/.h` as it was used only in hallu and dried hallu shrooms items, which are now obsoleted;
- Added comments on poison and magic shrooms and dried hallu shroom so they should be removed after 0.D;
- Added description of `STILL_POISONOUS` flag to `JSON_FLAGS.md`.

I found an issue with this system: batch-crafting (say, 4x) dried mushrooms, having 1 ordinary shroom and 3 hallucinogenic shrooms as components will result in 4 ordinary dried shrooms. Or recipes that require more than 1 mushrooms, one of which is ordinary and others are "special" will result in ordinary comestible, without any mushroom effects. AFAIK this is limitation of our crafting system as we cannot choose which component in question to use.

### CODE REVIEW

**Comment by nexusmrsep** [comment]
> let's presume proper cooking removes all toxins and poisons from a mushroom

IRL it does not, toxins & poisons are substances unlike living bacteria that die when heat processed, and while some of them get destroyed or changed on the molecular level via heating, many others don't. Where I live many people die yearly or suffer greatly from eating cooked or even fried poisonous mushrooms (because no one eats them raw) that were taken from the forest and were considered edible by mistake. It's not lack of knowledge (though sometimes it is) but sheer fact that some species of poisonous mushrooms are very similar to other edible mushrooms. Some can be discerned by examination, some even by taste or discoloration, but there are also some that cannot be properly discerned by organoleptic examination, especially when being still young and undeveloped.

**Comment by Night-Pryanik** [comment]
> while some of them get destroyed or changed on the molecular level via heating

Then let's presume `poisonous` mushrooms in Cata are from this ⬆️  kind of mushrooms. I think splitting them into "poisonous but edible after proper cooking" and "permanently poisonous" wouldn't be fun and isn't worth it.

File: data/json/items/comestibles.json [hunk 1]
```diff
@@ -3913,18 +3914,18 @@
     "price": 180,
     "material": "mushroom",
     "volume": 1,
-    "flags": [ "EDIBLE_FROZEN" ],
-    "charges": 4,
+    "flags": [ "EDIBLE_FROZEN", "STILL_POISONOUS" ],
+    "charges": 1,
```
Comment (ZhilkinSerg) [inline data/json/items/comestibles.json]: Charges is set to 1 by default, so you can omit it.

File: src/item_factory.cpp [hunk 2]
```diff
@@ -592,7 +592,6 @@ void Item_factory::init()
     add_iuse( "FLUSLEEP", &iuse::flusleep );
     add_iuse( "FLU_VACCINE", &iuse::flu_vaccine );
     add_iuse( "FUNGICIDE", &iuse::fungicide );
-    add_iuse( "FUN_HALLU", &iuse::fun_hallu );
```
Comment (ZhilkinSerg) [inline src/item_factory.cpp]: `FUN_HALLU` should also be removed from `item_actions.json`.

```JSON
    {
        "type" : "item_action",
        "id" : "FUN_HALLU",
        "name" : "Eat"
    },{
```

**Comment by sfsworms** [comment]
Disagree: if you can make all shrooms safe by cooking them you remove the point of having poisonous shrooms in the first place

**Comment by Night-Pryanik** [comment]
Someone might want to eat them raw. Someone might want to dry them for longer storage.

**Pull request merged**